src/Fantomas/CodePrinter.fs

```diff
@@ -867,14 +867,24 @@ and genExpr astContext synExpr =
     | Record(inheritOpt, xs, eo) ->
         let recordExpr =
             let fieldsExpr = col sepSemiNln xs (genRecordFieldName astContext)
+            let fieldsExprShortForm = (!- " with " +> fieldsExpr)
             eo |> Option.map (fun e ->
-                genExpr astContext e +> ifElseCtx (futureNlnCheck fieldsExpr) (!- " with" +> indent +> sepNln +> fieldsExpr +> unindent) (!- " with " +> fieldsExpr))
+                genExpr astContext e +> ifElseCtx (futureNlnCheck fieldsExprShortForm)
+                                                  (!- " with" +> indent +> sepNln +> fieldsExpr +> unindent)
+                                                  fieldsExprShortForm)
             |> Option.defaultValue fieldsExpr
 
+        let someSeparator =
```

Comment:
This is a bad name, but I think breaking up the code below a little is necessary.